### PR TITLE
PERF: use _get_axis_number instead of if axis in [...]

### PR DESCRIPTION
**Pull request by KiranHipparagi**
Added _get_axis_number in style.py as per the standards

### CODE REVIEW

**Comment by phofl** [comment]
A pr for this was already merged

**Comment by KiranHipparagi** [comment]
@phofl yes i know but there is one more change we need look at..
```
axis = self.data._get_axis_number(axis)            <-- PR
result = data.apply(func, axis=0, **kwargs) if axis == 0 
    else (data.T.apply(func, axis=0, **kwargs).T if axis == 1 
    else func(data, **kwargs))
       
 if axis in [0, "index"]:                           <--master
         result = data.apply(func, axis=0, **kwargs)
 elif axis in [1, "columns"]:
         result = data.T.apply(func, axis=0, **kwargs).T  # see GH 42005
 else:
     result = func(data, **kwargs)
```

**Comment by phofl** [comment]
Please resolve conflicts first

**Comment by KiranHipparagi** [comment]
@phofl thanks for the clarification.please review i have resolved conflicts.

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1149,27 +1149,26 @@ def _apply(
...
-                        f"Expected shape: {data.shape}"
-                    )
-                result = DataFrame(result, index=data.index, columns=data.columns)
-        else:
-            axis = self.data._get_axis_number(axis)
-            if axis == 0:
-                result = data.apply(func, axis=0, **kwargs)
-            else:
-                result = data.T.apply(func, axis=0, **kwargs).T  # see GH 42005
+
```
Comment (phofl) [inline pandas/io/formats/style.py:1152]: What is the benefit? This is harder to read imo

**Comment by attack68** [comment]
-1. I like it the way it is. As @phofl states this has already had a recent PR fix.

**Comment by KiranHipparagi** [comment]
@attack68 @phofl Thanks for the clarification.I'm new to open source project.Is there any file left for modification apart from style.py?

**Comment by attack68** [comment]
@KiranHipparagi did a quick search, it seems outside of tests, this is pretty clean actually. Do find another issue marked "good first issue".

**Pull request closed without merging**